File: web-service/yacontest-helper/src/main/java/org/ilyatyamin/yacontesthelper/grades/service/core/GradesService.java
package org.ilyatyamin.yacontesthelper.grades.service.core;

import org.ilyatyamin.yacontesthelper.grades.dto.GoogleSheetsRequest;
import org.ilyatyamin.yacontesthelper.grades.dto.GradesRequest;
import org.ilyatyamin.yacontesthelper.grades.dto.GradesRequestWithString;
import org.ilyatyamin.yacontesthelper.grades.dto.GradesResponse;

public interface GradesService {
    GradesResponse getGradesList(GradesRequest gradesRequest);

    GradesResponse getGradesList(GradesRequestWithString gradesRequest);

    byte[] generateGradesTable(Long tableId);

    void writeToGoogleSheets(Long tableId, GoogleSheetsRequest request);
}

File: web-service/yacontest-helper/src/main/java/org/ilyatyamin/yacontesthelper/grades/service/core/GradesService.kt
package org.ilyatyamin.yacontesthelper.grades.service.core

import org.ilyatyamin.yacontesthelper.grades.dto.GoogleSheetsRequest
import org.ilyatyamin.yacontesthelper.grades.dto.GradesRequest
import org.ilyatyamin.yacontesthelper.grades.dto.GradesResponse
import org.ilyatyamin.yacontesthelper.grades.enums.RequestType

interface GradesService {
    fun getGradesList(
        gradesRequest: GradesRequest?,
        requestType: RequestType = RequestType.BY_CLIENT
    ): GradesResponse

    fun generateGradesTable(tableId: Long?): ByteArray?

    fun writeToGoogleSheets(tableId: Long?, request: GoogleSheetsRequest?)
}
